src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Interop.Internal.Java.Util;
+using Microsoft.Spark.Interop;
+
+namespace Microsoft.Spark
+{
+    /// <summary>
+    /// A broadcast variable. Broadcast variables allow the programmer to keep a read-only variable
+    /// cached on each machine rather than shipping a copy of it with tasks.They can be used, for
+    /// example, to give every node a copy of a large input dataset in an efficient manner.Spark 
+    /// also attempts to distribute broadcast variables using efficient broadcast algorithms to 
+    /// reduce communication cost.
+    /// </summary>
+    [Serializable]
+    public sealed class Broadcast: IJvmObjectReferenceProvider
+    {
+        [NonSerialized]
+        private JvmObjectReference _jvmObject;
+        [NonSerialized]
+        private readonly SparkContext _sc;
+        [NonSerialized]
+        private string _path;
+        [NonSerialized]
+        private JvmObjectReference _python_broadcast;
+        private long _bid;
+
+        internal Broadcast(SparkContext sc,
+            object value,
+            JvmObjectReference sparkContext,
+            string path = null
+            )
+        {
+            if (sc != null)
+            {
+                // We're on the driver
```

Comment:
let's have separate constructors for each case

Reply:
Removed else as we don't need constructor for executor side.